pkg/cluster/cluster.go

```diff
@@ -481,6 +490,12 @@ func (c *Cluster) removePod(name string) error {
 		if !k8sutil.IsKubernetesResourceNotFoundError(err) {
 			return err
 		}
+		if c.cluster.Spec.SelfHosted != nil && c.debugLogger != nil {
+			c.debugLogger.LogMessage(fmt.Sprintf("pod (%s) not found while trying to delete it", name))
+		}
+	}
+	if c.cluster.Spec.SelfHosted != nil && c.debugLogger != nil {
```

Comment:
can you enable/disable the logger during the initialization? then we do not need to check this everywhere.